### PR TITLE
Improve documentation on intervals

### LINKED ISSUE #1518: [duckdb/#9643] - Issue #9631: Time/Interval Hours needs documentation
See https://github.com/duckdb/duckdb/issues/9643

### PR DESCRIPTION
**Pull request by szarnyasg**
Fixes #1518, only merge once https://github.com/duckdb/duckdb/pull/9643 hits `main`.

### CODE REVIEW

**Comment by szarnyasg** [comment]
@hawkfish is there anything else you'd like me to add to the documentation of intervals?

File: docs/sql/data_types/interval.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ SELECT DATE '2000-01-01' - INTERVAL 1 YEAR;
 SELECT INTERVAL (i) YEAR FROM range(1, 5) t(i);
 -- construct an interval with mixed units
 SELECT INTERVAL '1 month 1 day';
+-- intervals greater than 24 hours/12 months/etc. are supported
+SELECT ('25:00:00'::INTERVAL);
```
Comment (hawkfish) [inline docs/sql/data_types/interval.md:30]: Maybe make this three digits and non-zero? '987:65:43.210'?

**Review by hawkfish — COMMENTED** [review]
Otherwise looks good!

**Comment by szarnyasg** [comment]
@hawkfish thanks!

**Pull request merged**